src/components/workspace-utils.js

```diff
@@ -340,6 +342,46 @@ export const WorkspaceTagSelect = props => {
   })
 }
 
+export const WorkspaceStarControl = ({ workspace, starredWorkspaceIds, onUpdate, style }) => {
+  const [updating, setUpdating] = useState(false)
+
+  const { workspace: { workspaceId } } = workspace
+  const isStarred = _.includes(workspaceId, starredWorkspaceIds)
+
+  const toggleStar = _.flow(
+    Utils.withBusyState(setUpdating),
+    withErrorReporting(`Unable to ${isStarred ? 'unstar' : 'star'} workspace`)
+  )(async star => {
+    if (star) {
+      if (_.size(starredWorkspaceIds) < 50) {
+        const updatedWorkspaceIds = _.concat(starredWorkspaceIds, [workspaceId])
```

Comment:
Since one attribute contains all the starred workspaces, it would be good to reload the list immediately before updating. Currently, if you open two tabs and star workspaces in each one, the first one gets overwritten.

Reply:
Handled in 7deb3043e3c8496d8ab0a2363ad08706d21109b8

Note that I did make a small tweak to that code in https://github.com/DataBiosphere/terra-ui/pull/3229/commits/2a80a5abd17abe0b43cb61425e0ae30881a6219d